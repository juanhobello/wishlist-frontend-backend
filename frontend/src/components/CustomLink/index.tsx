import { ReactNode } from 'react';
import style from './CustomLink.module.css'
import { Link } from "react-router-dom";

interface CustomLinkProps {
  path: string;
  children: ReactNode;
}

export default function CustomLink({ path, children }: CustomLinkProps) {
  return (
    <Link className={style["link"]} to={path}>
      {children}
    </Link>
  )
}
