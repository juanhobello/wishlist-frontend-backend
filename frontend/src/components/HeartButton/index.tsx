import { FiHeart } from 'react-icons/fi'
import type { ButtonHTMLAttributes } from 'react'
import style from './HeartButton.module.css'

interface HeartButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  isActive: boolean
}

export default function HeartButton({ isActive, ...rest }: HeartButtonProps) {
  return (
    <button
      {...rest}
      className={`${style["heart-button"]} ${isActive ? style[".selected"] : ''}`}
      style={{ backgroundColor: isActive ? 'var(--heart-color)' : 'var(--gray-500)' }}
    >
      <FiHeart fontSize={20} />
    </button>
  )
}
