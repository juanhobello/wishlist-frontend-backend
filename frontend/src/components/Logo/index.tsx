import { Link } from "react-router-dom";
import styles from './Logo.module.css'

interface LogoProps {
  path: string;
}
export default function Logo({ path }: LogoProps) {
  return (
    <Link to={path} className={styles["logo-container"]} aria-label="Logo">
      <div className={styles["logo"]} />
    </Link>
  )
}
