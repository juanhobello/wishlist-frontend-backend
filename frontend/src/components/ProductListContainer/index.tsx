import type { ReactNode } from "react"
import style from './ProductListContainer.module.css'

interface ProductListContainerProps {
  children: ReactNode
}
export default function ProductListContainer({ children }: ProductListContainerProps) {
  return (
    <div className={style["products-list"]}>{children}</div>
  )
}
