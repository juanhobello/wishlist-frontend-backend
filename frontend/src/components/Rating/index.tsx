import styles from './Rating.module.css';
import Star from './Star';

interface RatingProps {
  rating: number
}

export default function Rating({ rating }: RatingProps) {

  const stars = Array.from({ length: 5 }, (_, i) => {
    const percentage = Math.min(Math.max(rating - i, 0), 1) * 100;
    return <Star key={i} percentage={percentage} />;
  });

  return (
    <div className={styles.rating}>
      {stars}
      <p>{rating.toFixed(1)}</p>
    </div>
  )
}

