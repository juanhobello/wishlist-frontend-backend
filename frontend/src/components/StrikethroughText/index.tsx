import type{ ReactNode } from "react"
import style from './StrikethroughText.module.css'

interface StrikethroughTextProps {
  children: string | ReactNode
}

export default function StrikethroughText({ children }: StrikethroughTextProps) {
  return (
    <s className={style["text"]}>{children}</s>
  )
}
