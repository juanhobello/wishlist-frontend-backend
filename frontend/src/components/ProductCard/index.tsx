import { Rating, StrikethroughText } from "@/components"
import type { ReactNode } from "react"
import styles from "./ProductCard.module.css"
import { Product } from "@/redux/types"
import { formatPrice } from "@/utils"

interface ProductCardProps {
  product: Product
  floatButton?: ReactNode
}
export default function ProductCard({ product, floatButton }: ProductCardProps) {
  return (
    <div className={styles["product-card"]}>
      {!!floatButton && <span className={styles["float-button"]}>{floatButton}</span>}

      <img src={product.image} className={styles["img"]} alt={product.name} />
     
      <div className={styles["title-product"]}>
        <p>{product.name}</p>
      </div>
     
      <Rating rating={product.rating} />

      <StrikethroughText>{formatPrice(Number(product.fullPriceInCents))}</StrikethroughText>

      <span className={styles["price"]}>
        {formatPrice(Number(product.salePriceInCents))}
      </span>
    </div>
  )
}
