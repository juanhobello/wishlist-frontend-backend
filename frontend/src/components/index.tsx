import Logo from "./Logo";
import CustomLink from './CustomLink'
import { Menu, MenuItem } from "./Menu";
import { Breadcrumbs, BreadcrumbItem } from "./Breadcrumbs";
import Divider from "./Divider";
import Rating from "./Rating";
import StrikethroughText from './StrikethroughText'
import ProductCard from "./ProductCard";
import ProductListContainer from "./ProductListContainer";
import HeartButton from './HeartButton';

export {
  Logo,
  CustomLink,
  Menu,
  MenuItem,
  Breadcrumbs,
  BreadcrumbItem,
  Divider,
  Rating,
  StrikethroughText,
  ProductCard,
  ProductListContainer,
  HeartButton
}