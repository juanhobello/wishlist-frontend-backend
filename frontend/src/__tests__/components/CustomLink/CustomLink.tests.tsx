import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { CustomLink } from "@/components";

const path = "/wishlist";

const setupTestCustomLink = () => (
  <BrowserRouter>
    <CustomLink path={path}>
      <span>Home</span>
    </CustomLink>
  </BrowserRouter>
);

describe("CustomLink component", () => {
  test("matches snapshot", () => {
    const { container } = render(setupTestCustomLink());
    expect(container).toMatchSnapshot();
  });

  test("renders Link with correct path", () => {
    render(setupTestCustomLink());

    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", path);
  });
})