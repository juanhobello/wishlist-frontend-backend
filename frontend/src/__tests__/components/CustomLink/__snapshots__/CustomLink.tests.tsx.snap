// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLink component matches snapshot 1`] = `
<div>
  <a
    class="link-wrapper"
    data-discover="true"
    href="/wishlist"
  >
    <span>
      Home
    </span>
  </a>
</div>
`;
